substrate/frame/revive/rpc/src/client/runtime_api.rs

```diff
@@ -65,8 +69,35 @@ impl RuntimeApi {
 	pub async fn dry_run(
 		&self,
 		tx: GenericTransaction,
+		block: BlockNumberOrTagOrHash,
 	) -> Result<EthTransactInfo<Balance>, ClientError> {
-		let payload = subxt_client::apis().revive_api().eth_transact(tx.into());
+		let timestamp_override = match block {
+			BlockNumberOrTagOrHash::BlockTag(BlockTag::Pending) =>
+				Some(Timestamp::current().as_millis()),
+			_ => None,
+		};
+
+		let payload = subxt_client::apis().revive_api().eth_transact_with_config(
+			tx.clone().into(),
+			DryRunConfig { timestamp_override }.into(),
+		);
+
+		// check if eth_transact_with_config is available
+		if let Err(err) = self.0.validate(&payload) {
```

Comment:
This adds one additional request every time. Can you try to just call the runtime API and match on the error? Hopefully you can detect if the fail was due to a missing runtime API.